src/constants/index.ts

```diff
@@ -16,4 +17,28 @@ export const HALLS = [
   'Sellards',
   'Stephenson',
   'Watkins',
-] as const;
```

Comment:
Why change back to the older implementation? The halls array should be a readonly export and it also simplifies the type definition for Hall.

Reply:
Personally I dislike typescript's readonly feature... since most values in an application aren't immutable it's kind of a pain to make the entire app work around a set of readonly values (`as` casting, spreading the value to make a mutable copy, etc). See https://github.com/resident51/resident51/pull/6/commits/c9df616e05612e89ad43755060869763f57d3ed5 - I added a comment in SignUp.tsx to show both a spread operator and an as cast as an example.

But readonly comes down to personal preference - would you like to keep it?